core/src/testils/criterion.rs

```diff
@@ -5,15 +5,22 @@ use std::cmp;
 use std::time::{Duration, Instant};
 
 pub trait BencherExt {
-    fn iter_batches<R, S, O>(&mut self, batch_size: usize, strategy: S, routine: R)
+    fn prop_iter_batched<R, S, O>(&mut self, batch_size: usize, strategy: S, routine: R)
     where
         R: FnMut(S::Value) -> O,
         S: Strategy;
 }
 
 impl BencherExt for Bencher<'_> {
-    fn iter_batches<R, S: Strategy, O>(&mut self, batch_size: usize, strategy: S, mut routine: R)
-    where
+    /// Similar to criterion's iter_batched fn, but allows for a batch, of
```

Comment:
the doc reads very confusing. maybe a bit simpler like "similar to iter_batched, but use a proptest strategy as the setup to randomly generate inputs".

also, maybe move the `batch_size` to the end to match the argument order of `iter_batched`.